### DISCUSSION TITLE
API log objects retrieval

### DISCUSSION THREAD

**Discussion by MaxWinsemius** [moby/moby#45846] [Q&A]
Is there a way to retrieve from the API the object logs as they are defined in `LogMessage`s? The [documentation](https://docs.docker.com/engine/api/v1.43/#tag/Container/operation/ContainerLogs) only notes  that a stream can be returned, but looking through [the code](https://github.com/moby/moby/blob/14de8ba4ead0cd414dc3bbdab2c88b97e9840bc0/api/types/backend/backend.go#L39) clearly shows there is a structure from which the stream is built.

**Comment by corhere** [comment]
No, there is not. What is your use case? You might be better served by using a logging driver which forwards container logs to a more fully featured log collector, such as fluentd.
